tests/test_geoprocessing.py

```diff
@@ -4833,3 +4833,21 @@ def test_geometry_field_mismatch(self):
 
         self.assertIn(
             "The fields and attributes for feature 0", str(cm.exception))
+
+    def test_raster_reduce(self):
+        """PGP: test raster_reduce can calculate a sum."""
+        array = numpy.ones((512, 512))
+        raster_path = os.path.join(self.workspace_dir, 'raster.tif')
+        _array_to_raster(array, -1, raster_path)
+
+        def sum_blocks(total, block): return total + numpy.sum(block)
+        spy_sum_blocks = unittest.mock.Mock(wraps=sum_blocks)
+        result = pygeoprocessing.raster_reduce(spy_sum_blocks, raster_path, 0)
+        self.assertEqual(result, 512*512)
+
+        for i, (_, (total, block), _) in enumerate(spy_sum_blocks.mock_calls):

```

Comment:
I think I get what this is doing, but maybe a comment would help make the intent clear?